templates/nginx/galaxy-main.j2

```diff
@@ -1,310 +1,314 @@
-			add_header Access-Control-Allow-Origin $http_origin;
-			add_header Access-Control-Allow-Methods 'GET';
-			add_header Access-Control-Expose-Headers 'Content-Length,Content-Range';
-			add_header Access-Control-Allow-Headers 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range';
-		}
-	}
-
-	location /_upload {
-		upload_store {{ galaxy_config['galaxy']['nginx_upload_store'] }};
-		upload_limit_rate 32k;
-		upload_store_access user:rw group:rw all:rw;
-		upload_pass_form_field "";
-		upload_set_form_field "__${upload_field_name}__is_composite" "true";
-		upload_set_form_field "__${upload_field_name}__keys" "name path";
-		upload_set_form_field "${upload_field_name}_name" "$upload_file_name";
-		upload_set_form_field "${upload_field_name}_path" "$upload_tmp_path";
-		upload_pass_args on;
-		upload_pass /_upload_done;
-	}
-
-	location /_upload_done {
-		set $dst /api/tools;
-		if ($args ~ nginx_redir=([^&]+)) {
-				set $dst $1;
-		}
-		rewrite "" $dst;
-	}
-
-	location /_job_files {
-		if ($request_method != POST) {
-				rewrite "" /api/jobs/$arg_job_id/files last;
-		}
-		upload_store {{ galaxy_config['galaxy']['nginx_upload_job_files_store'] }};
+    listen 443 default_server;
+    listen [::]:443 default_server;
+
+    error_page 404 /404.html;
+    error_page 502 /502.html;
+    error_page 503 /503.html;
+    error_page 504 /503.html;
+
+    location /404.html {
+        root /usr/share/nginx/html;
+        internal;
+    }
+    location /502.html {
+        root /usr/share/nginx/html;
+        internal;
+    }
+    location /503.html {
+        root /usr/share/nginx/html;
+        internal;
+    }
+    location /504.html {
+        root /usr/share/nginx/html;
+        internal;
+    }
+
+    # Move to remote storage, hopefully hammer local disk slightly less?
+    #client_body_temp_path  /data/dnb01/nginx_upload_store/;
+    client_body_buffer_size 1024m;
+
+    location / {
+        # This is the backend to send the requests to.
+        proxy_pass       http://galaxy;
+        proxy_set_header Host               $host;
+        proxy_set_header X-Forwarded-For    $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto  $scheme;
+        proxy_set_header Upgrade            $http_upgrade;
+        proxy_set_header Connection         $connection_upgrade;
+
+        # Throttle downloads. First 100Mb are free, after that 5Mb/s.
+        # users can move to FTP if they need / we can sort that out.
+        limit_rate_after 100m;
+        limit_rate 5m;
+
+        add_header X-Clacks-Overhead 'GNU James Taylor (@jxtx) Simon Gladman (@slugger70) Jim Johnson (@jj-umn)';
+    }
+
+    # Temporarily break downloading :(
+    location /api/libraries/datasets/download/ {
+        return 302 https://usegalaxy.eu;
+    }
+    #location /history/export_archive {
+    #    return 302 https://usegalaxy.eu;
+    #}
```

Comment:
I moved this out of the other location section. Nested locations are not supported imho, so this never worked imho. Maybe we can remove it alltogether.

Reply:
Temporarily broken for [6 years](https://github.com/usegalaxy-eu/infrastructure-playbook/commit/c220d7d263b4890f3c39bf34dca45251144672da) 😆 Do you recall the motivation for the decision?